components/clickup/clickup.app.mjs

```diff
@@ -1,305 +1,788 @@
       async options({
-        list, page,
+        workspaceId, page,
       }) {
-        const { tasks } = await this.getTasks({
-          list,
-          page,
+        const templates = await this.getTaskTemplates({
+          workspaceId,
+          params: {
+            page,
+          },
         });
-        return tasks.map((task) => ({
-          label: task.name,
-          value: task.id,
+
+        return templates.map((template) => ({
+          label: template.name,
+          value: template.id,
+        }));
+      },
+    },
+    customFields: {
+      type: "string",
+      label: "Custom Field",
+      description: "Select a custom field",
+      async options({ listId }) {
+        const fields = await this.getCustomFields({
+          listId,
+        });
+
+        return fields.map((field) => ({
+          label: field.name,
+          value: field.id,
         }));
       },
     },
+    priorities: {
+      type: "string",
+      label: "Priority",
+      description: "The level of priority",
+      options: Object.keys(constants.PRIORITIES),
+      default: "Normal",
+    },
   },
   methods: {
-    async _makeRequest({
-      method = "GET", endpoint, data, $,
-    }) {
+    /**
+     * Get the access token;
+     *
+     * @returns {string} The access token.
+     */
+    _accessToken() {
+      return this.$auth.oauth_access_token;
+    },
+    /**
+     * Get the base url of Asana API;
+     *
+     * @returns {string} The Asana Api base url.
+     */
+    _apiUrl() {
+      return "https://api.clickup.com/api/v2";
+    },
+    _headers() {
+      return {
+        Accept: "application/json",
+        Authorization: this._accessToken(),
+      };
+    },
+    /**
+     * Make a requests with pre defined options.
+     *
+     * @param {string} path - The path to make the request.
+     * @param {object} options - A default Axios options object.
+     * @param {object} $ - A default Pipedream run object.
+     *
+     * @returns {string} The request result data.
+     */
+    async _makeRequest(path, options = {}, $) {
       const config = {
-        headers: {
-          "content-type": "application/json",
-          "Authorization": this.$auth.oauth_access_token,
-        },
-        method,
-        url: `https://api.clickup.com/api/v2/${endpoint}`,
+        url: `${this._apiUrl()}/${path}`,
+        headers: this._headers(),
+        ...options,
       };
-      if (data) {
-        data = Object.entries(data).reduce((a, [
-          k,
-          v,
-        ]) => (v
-          ? (a[k] = v, a)
-          : a), {});
-        config.data = data;
-      }
-      const response = await axios($ || this, config).catch((err) => {
-        if (err.response.status !== 200) {
-          throw new Error(`API call failed with status code: ${err.response.status}`);
-        }
-      });
-      return response;
+
+      return await axios($ ?? this, config);
+    },
+
+    async getWorkspaces({ $ } = {}) {
```

Comment:
You can remove the empty lines between methods.
```suggestion
    },
    async getWorkspaces({ $ } = {}) {
```

Reply:
Is this dont a bad practice ? 

Because will dirty the code, make sense ?